lib/Relationship.php

```diff
@@ -602,7 +602,7 @@ public function create_association(Model $model, $attributes=array(), $guard_att
 	public function load_eagerly($models=array(), $attributes=array(), $includes, Table $table)
 	{
 		$this->set_keys($table->class->name);
-		$this->query_and_attach_related_models_eagerly($table,$models,$attributes,$includes,$this->foreign_key, $table->pk);
+		$this->query_and_attach_related_models_eagerly($table,$models,$attributes,$includes,$this->foreign_key, $this->primary_key);
```

Comment:
Ding ding: this rings a bell now. @koenpunt is right, we'll want a test for this.